server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java

```diff
@@ -56,15 +56,20 @@
   private final List<Scanner> scanners;
   private final Iterator<Entry<Key,Value>> iter;
 
+  public RecoveryLogsIterator(ServerContext context, List<Path> recoveryLogDirs,
+      boolean checkFirstKEy) throws IOException {
+    this(context, recoveryLogDirs, null, null, checkFirstKEy);
+  }
```

Comment:
It seems like this new constructor was added for convenience, but it doesn't have a javadoc to explain how it differs from the other constructor. If your goal is to make it crystal clear when you would use these additional parameters and when you don't need them, instead of using constructors here you could use static methods whose names are descriptive in the API.

So, instead of the following in the calling code:

```java
var iter = new RecoveryLogsIterator(context, dirs, start, stop, false);
var iter2 = new RecoveryLogsIterator(context, dirs, true);
```

it could look like:

```java
var iter = RecoveryLogsIterator.scanRange(context, dirs, start, stop, false);
var iter2 = RecoveryLogsIterator.scanAll(context, dirs); // don't think you need the boolean for this case
```

Then, you can make the constructor private (or package-private, if you need it visible for testing), and the implementing static methods can do some additional parameter validation to ensure that when you specify a range, both arguments aren't null.

These are just general design tips, though. For this, since the new constructor is only ever used once, I'd probably just remove it and pass in nulls on the old constructor for the one time it is called. Minimally, if you're keeping the new constructor, it should have a javadoc (and you should double check to see if you actually need to pass the boolean, since I don't think it matters when the first key is null).

Reply:
It is probably best to delete it and use passed in null here. The only reason I added it was to keep it consistent with how `RecoveryLogReader` did a rangeless read.